### ISSUE TITLE
"neighbors" in getNodeInfo RPC should not include disconnected neighbors

### ISSUE THREAD

**Issue by c0deright** [iotaledger/hornet#636]
**Describe the bug**

The `getNodeInfo` RPC call includes a return item called `neighbors`. With [iri](https://github.com/iotaledger/iri) the number returned always represented the number of all *connected* peers as in: you had 10 peers configured in iri but 3 were down so iri only had 7 successfull connections. The number returned by `getNodeInfo` RPC call for `neighbors` would be 7.

After migrating to hornet I did add some static peers in `peering.json`. Some of them are successfully connected by, some aren't.

The `getNodeInfo` RPC call returns the number of (number of currently connected peers) + (number of disconnected peers).

**To Reproduce**

- add static nodes to `peering.json`
- restart hornet
- run the following commands:

```
curl -sH 'X-IOTA-API-Version: 1' http://localhost:14265 -d '{"command": "'getNeighbors'"}' | jq '.neighbors[] | select(.connected==true) | .domain' | wc -l

curl -sH 'X-IOTA-API-Version: 1' http://localhost:14265 -d '{"command": "'getNeighbors'"}' | jq '.neighbors[] | select(.connected==false) | .domain' | wc -l

curl -sH 'X-IOTA-API-Version: 1' http://localhost:14265 -d '{"command": "'getNodeInfo'"}' | jq .neighbors
```

**Expected output**

```
10
6
10
```

**Actual output**

```
10
6
16
```

**Comment by muXxer** [comment]
You already filter by the connected flag in your commands, so you see that we changed the behavior to show all neighbors, with the flag if they are connected or not. There is nothing wrong with that.

**Comment by c0deright** [comment]
But what's the point in showing a number in the most prominent "get an 
overall status" RPC call that nobody can make any sense of?

getNodeInfo is perfect for monitoring and alerting if certain thresholds 
are reached, e.g. when a configured minimum amount of connected peers is 
reached.

With your change that number in getNodeInfo will always be arbitrary high 
even when not a single node is connected and someone should be notified. 
Thats pretty useless in my opinion.

For the monitoring solution to now get the real amount of connected peers 
one has to allow the getNeighborInfo RPC call when in the past the 
getNodeInfo call was enough and even then has to parse the json to filter 
for connected=true. Thats a lot of hassle for such a simple info "how many 
peers am I connected to".


Am 23. August 2020 14:25:53 schrieb muXxer <notifications@github.com>:

**Comment by muXxer** [comment]
@c0deright 
sorry I misread your issue report. I thought you are complaining that getNeighbors returns also disconnected neighbors, but you were talking about getNodeInfo. Sorry I missed that point.
OK valid issue, the question is now, do we change the count or do we add another field "connectedNeighbors" ?

**Comment by muXxer** [comment]
![grafik](https://user-images.githubusercontent.com/32371094/90980642-e5994300-e55c-11ea-83eb-9b3344f26226.png)
OK I will change the API to return the connected neighbors!
Thx for the issue report.

**Comment by c0deright** [comment]
Thanks! 👍🏻

**Comment by muXxer** [comment]
Fixed in #637
